src/univariate/continuous/pareto.jl

```diff
@@ -112,6 +112,12 @@ quantile(d::Pareto, p::Real) = cquantile(d, 1 - p)
 
 rand(rng::AbstractRNG, d::Pareto) = d.θ * exp(randexp(rng) / d.α)
 
+rand!(rng::AbstractRNG, d::Pareto, A::AbstractArray{<:Real}) =
+    # A .* 1/d.α is faster than A ./ d.α, but less exact
+    # fast-math style optimization; should be fine for PRNG
+    A .=  d.θ .* exp.(randexp!(rng, A) .* (1/d.α))
+
+
```

Comment:
I think using `randexp!` is an obvious improvement but I'm not sure whether `1/...` and the inconsistency between `rand` and `rand!` is worth it (generally `1/...` can easily lead to undesired promotions but I don't think that's a problem here). I'd use something like

```suggestion
xval(d::Pareto, z::Real) = d.θ * exp(z / d.α)

rand(rng::AbstractRNG, d::Pareto) = xval(d, randexp(rng))
function rand!(rng::AbstractRNG, d::Pareto, A::AbstractArray{<:Real})
    randexp!(rng, A)
    map!(Base.Fix1(xval, d), A, A)
    return A
end

```